### DISCUSSION TITLE
Getting the current iteration for a repeatable entry in Infolist

### DISCUSSION THREAD

**Discussion by sherifiskander** [filamentphp/filament#9216] [General]
I am trying to create a repeatable entry section that calls a function on the current array to generate the required text entries..

How can I find the current index of the repeater in infolist.

I am able to get the record using 

$inforecord = $infolist->getRecord();

but the in the repeater I need to know which iteration is the repeater on so I can call a function using $inforecord[INDEX]

**Comment by cakanm** [comment]
Have you found a solution for this?

**Comment by snarexx** [comment]
same here . Did you find a solution? @sherifiskander @cakanm

**Comment by bayprogrammer** [comment]
I ran into a need which I think is the same as this question. I was able to solve the problem (getting the data for the current iteration) like so:

```php
    Components\RepeatableEntry::make('profiles')
        ->schema([
            Components\ImageEntry::make('photo')
                ->label(function ($component) {
                    $index = explode('.', $component->getStatePath())[1];
                    $data = $component
                        ->getContainer()
                        ->getParentComponent()
                        ->getState()[$index];

                    return $data['first'] . ' ' . $data['last'];
                }),
            // ...
        ])
```

Got the hints I needed to work this out from the following other discussions:

* https://github.com/filamentphp/filament/discussions/4002
* https://github.com/filamentphp/filament/issues/8923#issuecomment-1761887812

In my case I couldn't just inject the model for the repeated entry as it's not backed by any model at all (it's array data).

**Comment by bilogic** [comment]
Is there a way to paginate a `RepeatableEntry`?

**Reply by francoism90** [reply]
@bilogic Did you ever found a solution? I also would like to paginate or at least limit the number of results.
